lib/src/msg.c

```diff
@@ -214,6 +214,7 @@ cc_msg_t* cc_msg_builder(int device_id, int command, const void *data_struct)
     }
     else if (command == CC_CMD_ASSIGNMENT)
     {
+        cc_device_t *device = data_struct;
```

Comment:
this is not quite right...
`data_struct` cannot be `cc_device_t` and `cc_assignment_t` at the same time.
what is going on here?